### ISSUE TITLE
juicefs metadata structure stored in engine has not considered scalability

### ISSUE THREAD

**Issue by liujiangang01** [juicedata/juicefs#3961]
**What would you like to be added**:
`	type Slice struct {
          id    uint64
	  size  uint32
	  off   uint32
	  len   uint32
	  pos   uint32
        }
`
Slice metadata structure has not considered scalability when been designed. If developer want to update Slice structure, it will cause a lot of trouble. I think adding a "version" field is nedded

**Comment by SandyXSD** [comment]
For now there's no requirement to add more field.

**Comment by liujiangang01** [comment]
it has no problem now. But store format is designed for now and future, it is very normal. For example, if we want to support new requirement for switching storage, Slice structure need changed. I think if we want to operate juicefs in a long time,  compatibility should be considerd in design

**Comment by SandyXSD** [comment]
Yes, we may add `version` field in the future together with new features, but won't do it now to just add the `version`.

**Comment by davies** [comment]
For compatibility, newly added field will have a default value (0, for example), we can use that for backward compatibility.
